packages/storage/amplify_storage_s3_dart/lib/src/storage_s3_service/transfer/database/transfer_record.dart

```diff
@@ -0,0 +1,48 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:convert';
+
+import 'package:json_annotation/json_annotation.dart';
+
+part 'transfer_record.g.dart';
+
+/// {@template amplify_storage_s3_dart.transfer_data}
+/// the type used by Storage S3 plugin upload operations when using s3 multipart upload.
+/// it is persisted to TransferDatabase.
+/// {@endtemplate amplify_storage_s3_dart.transfer_data}
+@JsonSerializable()
+class TransferRecord {
+  /// {@macro amplify_storage_s3_dart.transfer_data}
+  const TransferRecord({
+    required this.uploadId,
+    required this.objectKey,
+    required this.createdAt,
+  });
+
+  /// creates new [TransferRecord] object from a [json] map.
+  factory TransferRecord.fromJson(Map<String, dynamic> json) =>
+      _$TransferRecordFromJson(json);
+
+  /// creates new [TransferRecord] object from [jsonString].
+  factory TransferRecord.fromJsonString(String jsonString) {
+    return TransferRecord.fromJson(
+      jsonDecode(jsonString) as Map<String, dynamic>,
+    );
+  }
+
+  /// The multipart upload id.
+  final String uploadId;
+
+  /// The object key associated with the [uploadId].
+  final String objectKey;
+
+  /// Timestamp of [uploadId] creation.
+  final String createdAt;
```

Comment:
`json_serializable` can handle a `DateTime` here which will help clean up the DateTime handling throughout

Reply:
using String type because the drift table uses TextColumn for this field. 
I agree that the code would abe cleaner if using the DateTime her and hence I gave it a try.
Due to changing the drift table schema we need to bump the schema version and add MigrationStrategy. after making thses changes I feel like changing the field type from string to DateTime adds complexity to the native impl and am not convinced if we should do it. @HuiSF wdyt?

Reply:
I don't follow - in the Drift code you create a `TransferRecordsCompanion` from this anyway, so the table could remain a string.

Reply:
SQLite DB doesn't have a DateTime type, the values of date time can be stored as TEXT, REAL or INTEGER. If we prefer using DateTime in the library types, we could call `createdAt.toIso8601String()` passing value to `TransferRecordsCompanion` while creating a DB entry.

If we make  `createdAt` `DateTime` type in `TransferRecord` to let json serializer handle the type conversion, I guess it will generate something like

```
createdAt: json['createdAt'] == null
          ? null
          : DateTime.parse(json['createdAt'] as String),
```
So it should still be compatible with the string value read from SQLite db?

Reply:
I thought we want to cleanup the DateTime handling and use DateTime type instead of string throughout. means updating the `TextColumn createdAt` to `DateTimeColumn createdAt` in tables.dart and use the drift builder option to handle the date time values as text in build.yaml
```
builders:
      drift_dev:
        options:
          store_date_time_values_as_text: true
```
per latest comments we want to just use DateTime for the `TransferRecord` and keep the drift table as is and convert DateTime to String for drift impl.

Reply:
I'm most concerned about having conversion logic interspersed with the business logic, e.g. https://github.com/aws-amplify/amplify-flutter/pull/2631/files#diff-842ab00c5bcfb4a64fb7459e3815b98ad02db3442f22e2a0d01e2cf4dce20f1aR48

Moving the conversion logic to the Drift code and/or changing the table type make sense to me, but the latter is most important.